src/index.js

```diff
@@ -16,6 +17,35 @@ const logger = {
 const baseURL = "https://secure.materiel.net";
 
 class MaterielnetKonnector extends CookieKonnector {
+    async fetch(fields) {
+        try {
+            // Try classic execution
+            await this.tryFetch(fields);
+        } catch (err) {
+            if (err.isCaptcha && err.isCaptcha === true) {
+                const $ = cheerio.load(err.body);
+                const websiteKey = $(".g-recaptcha").data("sitekey");
+                const websiteURL = err.url;
+                // Solve captcha
+                const captchaToken = await solveCaptcha({ websiteURL, websiteKey });
+                // End login
+                await this.login(err.loginToken, fields, captchaToken);
+                try {
```

Comment:
Maybe we could pass an optional parameter named `retry` default to `true` by default to `this.fetch` and reuse the function here by passing `retry` to false?

Reply:
I'm not sure of what you mean here. What kind of retry are you talking about ? A promise retry and cactch block ?

I'm comfortable with this kind of code, but if you want we can change it.

Reply:
I just thought this was doing pretty much the same thing than line 22 so we could do:

```js
    async fetch(fields, retry) {
        try {
            // Try classic execution
            await this.tryFetch(fields);
        } catch (err) {
            if (err.isCaptcha === true) {
                if (retry === false)
                    throw new Error(errors.CAPTCHA_RESOLUTION_FAILED);
                else {
                    // remaining code
```